### PR TITLE
Versioning script

### PR DESCRIPTION
**Pull request by chayim**
As per our discussion on #22 this is the first part of versioning, as discussed.

### CODE REVIEW

**Comment by rafie** [comment]
@chayim Can we please please please do `bash` instead of `sh`? It really hearts me in my good eye. :-)

**Comment by rafie** [comment]
@chayim I'm aware of the advantages of using `sh`, but I deliberately decided to use `bash` across the board because `sh` syntax is very poor. When considering the tradeoff (of having to explicitly install `bash` on alpine and to cope with `bash 3.0` on older macOS versions), I clearly prefer the possibility to use a reasonable syntax and of course `bashdb`, which is a major improvement in productivity.

**Comment by chayim** [comment]
Helping with tooling, I totally get. I'm biased to lowest common denominator - so I'll continue writing SH compliant Bash whenever possible. #! change inbound.

Let me know what you think of the function wrapper - and being forced to call the script with any argument (and ignoring it). The reason was to make the script itself sourceable - though we should yank that into shibumi.  Personally - I need to call it as a script via make.

File: bin/generateversion [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#    PRODUCT=$(basename `git rev-parse --show-toplevel`)
+#fi
+#
+if [ -z "${VERSION}" ]; then
+    VERSION=$(git branch|grep '*'|cut -d '*' -f 2-2|sed 's/^ *//g')
+fi
+
+# We change to the script directory, purely to call platform from the exact path
+HERE=`dirname $0`
+cd ${HERE}
```
Comment (rafie) [inline bin/generateversion:26]: If we source the script, do we need to protect the calling script from this `cd`?

**Comment by rafie** [comment]
@chayim Good news is that we probably can use `bashdb` to debug `sh` scripts. However, if we want to go on using `shibumi` goodies with `sh` then we're out of luck.
The bigger problems I find with going with `sh` are:
- A mixed `bash`/`sh` environment is looking for trouble,
- Writing proper `sh` requires more diligence that most people in this group lack. I don't want people to make stupid errors and we chasing them around.
So let's think about it a bit and see if it is worth it. Up until now, I haven't encounter any real/practical problem of doing `bash` even with the `3.2` language level (I avoid `5.0` syntax altogether).

**Comment by chayim** [comment]
Totally. It's a style thing - I try to write most common denominator until I have no choice.

File: bin/generateversion (hunk 1 above)
Comment (chayim) [reply bin/generateversion:27]: So I need the CD to cross the repository boundary as the repos are included. The CD only exists to call into platform. In the termination case, the script context ends and you're in the same directory. In the inclusion case - not.  pushd/popd can solve that - pushed it upstream, see how you like it.

**Pull request merged**